fe/fe-core/src/main/java/com/starrocks/scheduler/history/TableBasedTaskRunHistory.java

```diff
@@ -0,0 +1,156 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.scheduler.history;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonParser;
+import com.starrocks.common.util.DateUtils;
+import com.starrocks.load.pipe.filelist.RepoExecutor;
+import com.starrocks.scheduler.persist.TaskRunStatus;
+import com.starrocks.scheduler.persist.TaskRunStatusChange;
+import com.starrocks.statistic.StatsConstants;
+import com.starrocks.thrift.TResultBatch;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.logging.log4j.util.Strings;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.text.MessageFormat;
+import java.time.ZoneId;
+import java.util.List;
+
+/**
+ * History storage that leverage a regular starrocks table to store the data.
+ * By default, using the time-based garbage collection strategy, which keep 7days' history
+ */
+public class TableBasedTaskRunHistory implements TaskRunHistory {
+
+    public static final String DATABASE_NAME = StatsConstants.STATISTICS_DB_NAME;
+    public static final String TABLE_NAME = "task_run_history";
+    public static final String TABLE_FULL_NAME = DATABASE_NAME + "." + TABLE_NAME;
+    public static final int TABLE_REPLICAS = 3;
+    public static final String CREATE_TABLE =
+            String.format("CREATE TABLE IF NOT EXISTS %s (" +
+                    // identifiers
+                    "task_id bigint NOT NULL, " +
+                    "task_run_id string NOT NULL, " +
+                    "create_time datetime NOT NULL, " +
+                    "task_name string NOT NULL, " +
+
+                    // times
+                    "finish_time datetime NOT NULL, " +
+                    "expire_time datetime NOT NULL, " +
+
+                    // content in JSON format
+                    "history_content_json JSON NOT NULL" +
+                    ")" +
+
+                    // properties
+                    "PRIMARY KEY (task_id, task_run_id, create_time) " +
+                    "PARTITION BY RANGE(create_time)() " +
+                    "DISTRIBUTED BY HASH(task_id) BUCKETS 8 " +
+                    "PROPERTIES(" +
+                    "'replication_num' = '1'," +
+                    "'dynamic_partition.time_unit' = 'DAY', " +
+                    "'dynamic_partition.start' = '-7', " +
+                    "'dynamic_partition.end' = '3', " +
+                    "'dynamic_partition.prefix' = 'p' " +
+                    ") ", TABLE_FULL_NAME);
+
+    private static final String COLUMN_LIST = "task_id, task_run_id, task_name, " +
+            "create_time, finish_time, expire_time, " +
+            "history_content_json ";
+    private static final String INSERT_SQL_TEMPLATE = "INSERT INTO {0} " +
```

Comment:
use COLUMN_LIST to construct, do not use hard code, which is hard to change, and error-prone